### ISSUE TITLE
reading OME XML crashes when pyQtGraph has been imported

### ISSUE THREAD

**Issue by fevemo** [tlambert03/ome-types#261]
I am trying to use the ome_types package to read fluorescence lifetime (FLIM) images and display them within a GUI application. However, when the pyQtGraph package has been imported, xsdata crashes after calling either from_tiff or from_xml (see error below). I created a simple test script that replicates the behavior:

``` python
  from ome_types import from_tiff, from_xml
  import pyqtgraph as pg
  from tifffile import TiffFile
  
  file_path = "path/to/file.ome.tiff"

  # Option 1
  ome_data = from_tiff(file_path)
  # Option 2
  with TiffFile(file_path) as tif:
      ome_data = from_xml(tif.ome_metadata)

```

This was done with the following environment (python 3.11.0):
- annotated-types==0.7.0
- numpy==2.1.1
- ome-types==0.5.1.post1
- pydantic==2.9.2
- pydantic-compat==0.1.2
- pydantic_core==2.23.4
- PyQt5==5.15.11
- PyQt5-Qt5==5.15.2
- PyQt5_sip==12.15.0
- pyqtgraph==0.13.7
- tifffile==2024.9.20
- typing_extensions==4.12.2
- xsdata==24.3.1 (Installed automatically with ome-types)

The error can be fixed by modifying xsdata to prevent it iterating over all subclasses of object. E.g. by commenting out lines 548-549 in xsdata\formats\dataclass\parsers\nodes\element.py:

``` python
        if var.process_contents != "skip":
            clazz = self.context.find_type(qname)
```

Is there a way to prevent this by using a different parser or is this inherent to the way ome_types works? I would like to keep using ome_types because I also plan on writing new ome tiffs from python and I don't want to reinvent the wheel if possible.

```
ome_data = from_tiff(os.path.join(folder, filename))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\ome_types\_conversion.py", line 159, in from_tiff
    return from_xml(xml, validate=validate, parser_kwargs=parser_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\ome_types\_conversion.py", line 132, in from_xml
    return parser.parse(xml_2016, OME_type)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\xsdata\formats\dataclass\parsers\bases.py", line 56, in parse
    result = handler.parse(source)
             ^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\xsdata\formats\dataclass\parsers\handlers\native.py", line 43, in parse
    return self.process_context(ctx)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\xsdata\formats\dataclass\parsers\handlers\native.py", line 57, in process_context
    self.parser.start(
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\xsdata\formats\dataclass\parsers\bases.py", line 92, in start
    child = item.child(qname, attrs, ns_map, len(objects))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\xsdata\formats\dataclass\parsers\nodes\element.py", line 453, in child
    node = self.build_node(qname, var, attrs, ns_map, position)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\xsdata\formats\dataclass\parsers\nodes\element.py", line 549, in build_node
    clazz = self.context.find_type(qname)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\xsdata\formats\dataclass\context.py", line 166, in find_type
    types: List[Type] = self.find_types(qname)
                        ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\xsdata\formats\dataclass\context.py", line 151, in find_types
    self.build_xsi_cache()
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\xsdata\formats\dataclass\context.py", line 109, in build_xsi_cache
    if self.is_binding_model(clazz):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\xsdata\formats\dataclass\context.py", line 129, in is_binding_model
    if not self.class_type.is_model(clazz):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\Anaconda3\envs\omeTest\Lib\site-packages\xsdata_pydantic_basemodel\compat.py", line 97, in is_model
    if issubclass(clazz, BaseModel) and clazz != BaseModel:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen abc>", line 123, in __subclasscheck__
TypeError: unhashable type: 'ProxyMetaclass'
```

I can also provide a ome.tiff file to test if needed but I don't expect the behavior to depend on the type of image that is loaded.

**Comment by tlambert03** [comment]
oddly enough, I'm unable to reproduce with the same set of dependencies and python 3.11

<details>

```python
dev/self/ome-types on  tags/v0.5.1.post1 [?] via 🅒 ome-types
➜ python --version
Python 3.11.10

dev/self/ome-types on  tags/v0.5.1.post1 [?] via 🅒 ome-types
➜ pip list
Package                    Version                   Editable project location
-------------------------- ------------------------- --------------------------------
annotated-types            0.7.0
appdirs                    1.4.4
attrs                      24.2.0
babel                      2.16.0
certifi                    2024.8.30
cffi                       1.17.1
cfgv                       3.4.0
charset-normalizer         3.3.2
click                      8.1.7
click-default-group        1.2.4
colorama                   0.4.6
coverage                   7.6.1
cssselect                  1.2.0
decorator                  5.1.1
distlib                    0.3.8
docformatter               1.7.5
elementpath                4.5.0
filelock                   3.16.1
flexcache                  0.3
flexparser                 0.3.1
ghp-import                 2.1.0
griffe                     1.3.1
identify                   2.6.1
idna                       3.10
iniconfig                  2.0.0
jinja2                     3.1.4
jsonschema                 4.23.0
jsonschema-specifications  2023.12.1
lxml                       5.3.0
markdown                   3.7
markupsafe                 2.1.5
mergedeep                  1.3.4
mkdocs                     1.6.1
mkdocs-autorefs            1.2.0
mkdocs-get-deps            0.2.0
mkdocs-material            9.5.36
mkdocs-material-extensions 1.3.1
mkdocstrings               0.26.1
mkdocstrings-python        1.11.1
mypy                       1.11.2
mypy-extensions            1.0.0
nodeenv                    1.9.1
numpy                      2.1.1
ome-types                  0.5.1.post2.dev5+g2d48e90 /Users/talley/dev/self/ome-types
packaging                  24.1
paginate                   0.5.7
pathspec                   0.12.1
pint                       0.24.3
pip                        24.2
platformdirs               4.3.6
pluggy                     1.5.0
pre-commit                 3.8.0
pycparser                  2.22
pydantic                   2.9.2
pydantic-compat            0.1.2
pydantic-core              2.23.4
pygments                   2.18.0
pymdown-extensions         10.10.1
pyqt5                      5.15.11
pyqt5-qt5                  5.15.15
pyqt5-sip                  12.15.0
pyqtgraph                  0.13.7
pytest                     8.3.3
pytest-codspeed            2.2.1
pytest-cov                 5.0.0
pytest-mypy-plugins        3.1.2
python-dateutil            2.9.0.post0
pyyaml                     6.0.2
pyyaml-env-tag             0.1
referencing                0.35.1
regex                      2024.9.11
requests                   2.32.3
rpds-py                    0.20.0
ruff                       0.3.0
setuptools                 74.1.2
six                        1.16.0
tifffile                   2024.9.20
tomlkit                    0.13.2
toposort                   1.10
types-beautifulsoup4       4.12.0.20240907
types-html5lib             1.1.11.20240806
types-lxml                 2024.9.16
typing-extensions          4.12.2
untokenize                 0.1.1
urllib3                    2.2.3
virtualenv                 20.26.5
watchdog                   5.0.2
wheel                      0.44.0
xmlschema                  2.4.0
xsdata                     24.2.1
```

</details>

(I used the ome.tiff file in the test data here, in case it's file-specific... which seems unlikely, but possible)

i haven't tried windows yet (am on macos)

**Comment by fevemo** [comment]
Hey @tlambert03, thank you for the super quick reply! Surprisingly, it is file-specific. I just tested it out on a mac and with [my ome-tiff](https://drive.google.com/file/d/13WF0m07tpUD4cDX52V_KCfQt1ZRorZNN/view?usp=drive_link) file it fails while with the one here in tests/data/ome.tiff it doesn't.  

Our ome-tiffs are also 'homemade' using a custom C++ data acquisition library and the [ome-files library](https://gitlab.com/codelibre/ome/ome-files-cpp) but since they work fine when pyqtgraph is not there (and in FLIMfit and ImageJ) I never really though they could be the source of the issue. However, I wouldn't be surprised if they are not fully compliant to the OME standard.

I uploaded the file to a [google drive](https://drive.google.com/file/d/13WF0m07tpUD4cDX52V_KCfQt1ZRorZNN/view?usp=drive_link) so you can give it a try, maybe it has to do with the custom annotations or the modulo?

Thank you for the support!

**Comment by tlambert03** [comment]
Thanks, with that file I'm able to reproduce.  when I comment out the pyqtgraph import it does give a warning about one of the Annotation ID's in the file `Casting invalid AnnotationID '' to 'Annotation:18'` ... so it does appear to have a slight non-compliance.  But that shouldn't cause a full failure. So this is a helpful test case.  Will push a fix soon

**Comment by tlambert03** [comment]
should be fixed in v0.5.2, which is now on pypi.  Let me know if not!

**Comment by fevemo** [comment]
> should be fixed in v0.5.2, which is now on pypi. Let me know if not!

Works like a charm! Thanks again!
